viewing_party/party.py

```diff
@@ -1,23 +1,169 @@
+    
+    return average
 
+def get_most_watched_genre(user_data):
+    movie_genre = []
+    popular_genre={}
+
+    if len(user_data["watched"]) >0:
+    
+        for movie in user_data["watched"]:
+            popular_genre[movie["genre"]]=0
+            movie_genre.append(movie["genre"])
+
+        for genre in movie_genre:
+            if genre in popular_genre:
+                popular_genre[genre]+=1
+
+        most_popular_genre = max(popular_genre, key=popular_genre.get)
+
+        return most_popular_genre
+        
+    
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
 
-        
+def helper_function_friend (user_data):
+    friend_list =[]
+
+    for friend_data in user_data["friends"]:
+        for friend in friend_data["watched"]:
+            friend_list.append(friend)
+    return friend_list
+
+def get_unique_watched(user_data):
+    user_list = []
+    friend_list =[]
+
+    user_watched = user_data["watched"]
+    for movie in user_watched:
+        user_list.append(movie)
+
+    friend_data = helper_function_friend(user_data)
+    
+    for movie in friend_data:
+        if movie in user_list:
+            user_list.remove(movie)
+    return user_list  
+
+
+def get_friends_unique_watched(user_data):
+    user_list = []
+    unique_list = []
+
+    user_watched = user_data["watched"]
+    for movie in user_watched:
+        user_list.append(movie)
+
+    friend_data = helper_function_friend(user_data)
+
+    for movie in friend_data:
+        if movie not in user_list and movie not in unique_list :
+            unique_list.append(movie)
+    return unique_list
+
+
+
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
 
+
+def get_available_recs(user_data):
+    
+    recommendation_list = []
+
+    #passing in last function as a helper function :) 
+    friend_unique_watched = get_friends_unique_watched(user_data)
+
+    for movie_host in friend_unique_watched:
+        if movie_host["host"] in user_data["subscriptions"]:
+            recommendation_list.append(movie_host)
+    return recommendation_list
+    
+
 # -----------------------------------------
-# ------------- WAVE 5 --------------------
+# ------------- WAVE 5--------------------
 # -----------------------------------------
 
+
+def get_new_rec_by_genre(user_data):
+    
+    genre_list = []
+    max_genre = get_most_watched_genre(user_data)  
+    friend_unique_watched = get_friends_unique_watched(user_data)
+
+    for movie in friend_unique_watched:
+        if movie["genre"] == max_genre: 
+            genre_list.append(movie)
+    return genre_list
+
+
+def get_rec_from_favorites(user_data): 
+
+    user_list = []
+    recommended_list = []
+
+    user_watched = user_data["watched"]
+    for movie in user_watched:
+        user_list.append(movie)

```

Comment:
It looks like `user_list` and the loop that fills it isn't used to get the list of recommendations. If that variable isn't meaningfully used, we want to remove the code associated with it since it creates a place where bugs could crop up.